src/odin/adapters/graph_adapter.py

```diff
@@ -61,28 +68,43 @@ def toJSON(self):
 
 class AvgGraphDataset(GraphDataset):
     
-    def __init__(self, time_interval, retention, name, source):
-        super().__init__(time_interval, adapter=None, get_path=None, retention=retention, name=name)
+    def __init__(self, time_interval, retention, source, location):
+        super().__init__(time_interval, adapter=None, get_path=None, retention=retention, location=location)
         
         self.source = source
         self.num_points_get = int(self.time_interval / self.source.time_interval)
 
-        logging.debug("This is an averaging dataset, averaging from %s", self.source)
+        self.min_list = []
+        self.max_list = []
+
+        logging.debug("Created averaging dataset " + location + ", averaging from %s", self.source)
 
     def get_data(self):
-        cur_time = time.time()
+        cur_time = time.strftime("%H:%M:%S", time.localtime())
         data = self.source.data[-self.num_points_get:]  # slice, get last x elements
+
+        data_min = min(data)
+        data_max = max(data)
+
         # data = list(zip(*data))[1]  # zip the timestamps and data of target list into separate
         data = data = sum(data) / len(data)
 
-        # logging.debug(data)
+        self.min_list.append(data_min)
+        self.max_list.append(data_max)
 
         self.data.append(data)
         self.timestamps.append(cur_time)
+
         if len(self.data) > self.retention:
             self.data.pop(0)
             self.timestamps.pop(0)
 
+            self.min_list.pop(0)
+            self.max_list.pop(0)
+
+        self.min = min(self.min_list)
+        self.max = max(self.max_list)
```

Comment:
This feels redundant? why not just keep a single Min and Max value and check each time if the new value is more/less, then update it? If you need it to be the min/max of only whats currently visible, then you can just do what you've done above and do min(data)